### ISSUE TITLE
[REVIEW]: DataLad: distributed system for joint management of code, data, and their relationship

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3262]
**Submitting author:** @yarikoptic (<a href="http://orcid.org/0000-0003-3456-2493">Yaroslav Halchenko</a>)
**Repository:** <a href="https://github.com/datalad/datalad" target ="_blank">https://github.com/datalad/datalad</a>
**Version:** 0.14.3
**Editor:** @arokem
**Reviewer:** @szorowi1, @jkanche
**Archive:** <a href="https://doi.org/10.5281/zenodo.5034875" target="_blank">10.5281/zenodo.5034875</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/be6dfc853d7993c156456a6450b24dbc/status.svg)](https://joss.theoj.org/papers/be6dfc853d7993c156456a6450b24dbc)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/be6dfc853d7993c156456a6450b24dbc"><img src="https://joss.theoj.org/papers/be6dfc853d7993c156456a6450b24dbc/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/be6dfc853d7993c156456a6450b24dbc/status.svg)](https://joss.theoj.org/papers/be6dfc853d7993c156456a6450b24dbc)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/datalad/datalad" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@szorowi1 & @jkanche, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @arokem know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/datalad/datalad">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@yarikoptic) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/datalad/datalad">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@yarikoptic) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
PDF failed to compile for issue #3262 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch paper-joss. Reticulating splines etc...
```

**Comment by arokem** [comment]
Question for @yarikoptic and @mih: I see that you are both "co-first author", but one of you is also the last author. I am not sure how that's supposed to work. Could you please explain what you intended here?

**Comment by yarikoptic** [comment]
"Contributed equally" would be more appropriate indeed given current authors ordering.  Can/should we adjust manuscript and re-render here right away?

**Comment by yarikoptic** [comment]
cool, so we don't forget I just committed/pushed to the [original repo](http://github.com/datalad/datalad-paper-joss) for now

**Comment by jkanche** [comment]
@arokem seems like I need to be assigned to check things on my review list

**Comment by danielskatz** [comment]
@Whedon re-assign @jkanche as reviewer

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by danielskatz** [comment]
@Whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by danielskatz** [comment]
@whedon re-invite @jkanche as reviewer

**Comment by whedon** [comment]
The reviewer already has a pending invite.

@jkanche please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by jkanche** [comment]
thank you @danielskatz! I am now able to review.

**Comment by mih** [comment]
Hey! Thanks in advance to the reviewers for wading through our code and docs -- certainly not a quick thing. I would nevertheless like to ask if there is anything we can do to streamline the process. I am only aware of one issue having been reported https://github.com/datalad/datalad/issues/5685

Have @jkanche @szorowi1 found something that we could proactively tackle?

Thx and best!

**Comment by arokem** [comment]
👋 @szorowi1 : have you had a chance to take a look?

**Comment by szorowi1** [comment]
Very sorry for the delay! I've worked through my review.

It's a very impressive piece of software. I've read through much of the documentation and several of the helpful tutorial videos. I haven't been able to test every aspect of its functionality, but I've successfully been able to install Datalad and work through the tutorials. I especially like the conversational style of the docs -- makes it very easy to read and follow.

I have no quibbles. It's very clear how much effort has gone into the documentation and software to make it user-friendly and easy-to-use.

**Comment by arokem** [comment]
Thanks @szorowi1! 

@jkanche : I see that there are only two remaining check boxes in your list. Do you have any comments regarding the description of the state of the art or references included (or not included) in the article?

**Comment by jkanche** [comment]
This is really cool and I 100% agree with @szorowi1. I've been using it and it's been a great experience so far. The documentation is very thorough and has been super helpful to understand how things work. 

Thanks for the awesome work!

**Comment by yarikoptic** [comment]
Thank you @szorowi1 @jkanche and @arokem !  Meanwhile, we ran into a paper which might be giving even a better presentation to the DataLad that we had prepared for JOSS https://psyarxiv.com/w8trm .  I assume we still have a possibility to adjust the manuscript with more references?

**Comment by mih** [comment]
Wonderful! Thanks @jkanche and @szorowi1 for your assessment and the time and energy spent to wade through the materials!

**Comment by arokem** [comment]
Thanks @jkanche and @szorowi1! 

As the reviewers have approved the submission, the next steps are for me to give this one more read, to make sure that there are no errors or issues. Then, we'll make sure that the metadata for the software/paper match and pass this on to approval by the EiCs. I'll provide guidance at every step.

For now, @yarikoptic : yes - you can add more references. Could you please add the reference now, and then let me know when you are ready for me to give it a read through?

**Comment by yarikoptic** [comment]
@whedon generate pdf from branch paper-joss

@arokem pushed, but give us till tomorrow to finalize , and we will buzz you. Thank you

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch paper-joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03262/joss.03262/10.21105.joss.03262.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03262/joss.03262/10.21105.joss.03262.pdf) :page_facing_up: :point_left:

**Comment by yarikoptic** [comment]
Sorry whedon, you were faster than my fingers. Try now

@whedon generate pdf from branch paper-joss

**Comment by mih** [comment]
@whedon generate pdf from branch paper-joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch paper-joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03262/joss.03262/10.21105.joss.03262.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03262/joss.03262/10.21105.joss.03262.pdf) :page_facing_up: :point_left:

**Comment by mih** [comment]
@whedon generate pdf from branch paper-joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch paper-joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03262/joss.03262/10.21105.joss.03262.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03262/joss.03262/10.21105.joss.03262.pdf) :page_facing_up: :point_left:

**Comment by mih** [comment]
I have now updated the manuscript with the following changes:

- spelling fixes
- add citation for another system that uses datalad
- a highres version of figure 2
- list the datalad UKBiobank extension too

It should hopefully be acceptable for publication now. Thx for given us the opportunity for these last-minute updates.

**Comment by yarikoptic** [comment]
a promised (but forgotten to be done) ping @arokem . Cheers and thank you !

**Comment by arokem** [comment]
Hello! Overall, looks good. I have a couple of small suggestions: 

1. I think that the section title "Overview of the DataLad and its ecosystem"  should be "Overview of DataLad and its ecosystem" or "Overview of the DataLad software and its ecosystem" (I would prefer the former), unless there is some intention in this wording that I am missing. 

2. https://github.com/datalad/datalad/pull/5766

**Comment by yarikoptic** [comment]
Thank you @arokem!
1. took the former (removing "the")
2. included the change to bib - not sure how it came to be `@misc` ;)

**Comment by yarikoptic** [comment]
@whedon generate pdf from branch paper-joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch paper-joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03262/joss.03262/10.21105.joss.03262.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03262/joss.03262/10.21105.joss.03262.pdf) :page_facing_up: :point_left:

**Comment by arokem** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon recommend-accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by arokem** [comment]
@whedon check references

**Comment by arokem** [comment]
@whedon check references from branch paper-joss

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch paper-joss
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1515/nf-2020-0037 is OK
- 10.5281/zenodo.4773629 is OK
- 10.5281/ZENODO.2431914 is OK
- 10.5281/ZENODO.2558512 is OK
- 10.5281/ZENODO.3874225 is OK
- 10.5281/ZENODO.3900277 is OK
- 10.5281/ZENODO.4495560 is OK
- 10.1038/sdata.2016.18 is OK
- 10.1038/s41467-021-21970-2 is OK
- 10.21203/rs.3.rs-264855/v1 is OK
- 10.1038/sdata.2016.44 is OK
- 10.31234/osf.io/w8trm is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by arokem** [comment]
OK! At this point could you:

- [ ] Make a tagged release of your software and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check that the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

We can then move forward with a final check by one of the EiC, before accepting the paper.

**Comment by yarikoptic** [comment]
Just to make sure - paper branch can remain detached (not merged into) from a release branch (maint) and tag, correct?

**Comment by arokem** [comment]
I believe that's correct, but let's bring in @openjournals/joss-eics to help resolve this. 

This is a case where the paper is on a separate branch. The release version that gets tagged and archived does not have to include this branch in it, does it?

**Comment by arfon** [comment]
> Just to make sure - paper branch can remain detached (not merged into) from a release branch (maint) and tag, correct?

Yes, that's right. We can accept the submission without the paper actually _landing_ in the merged code into the repository.

**Comment by yarikoptic** [comment]
> * Check that the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it).

Does it require to have author list in the same **order** in this paper and on Zenodo, or just that the "set" of authors match?

**Comment by yarikoptic** [comment]
While revisiting the list of the requirements for Zenodor archive entry, another concern is

> ... make sure the list is correct and people who only made a small fix are not on it

- we have contributors who contributed no "fix" at all, but were significant and notable contributors to see to the success of the project.  I think this requirement/statement should be adjusted for JOSS to not suggest a demand only for major "in commit/code" contributions: projects do often need administrators, mentors, user support,   and other "personnel" which would not be submitting a "fix" but would be critical.
- our zenodo entry might have (didn't check) minor "small fix" contributors to the software which decided to not become a co-author to the paper (and rather just to be acknowledged in Acknowledgements section).  I would hate to remove them from the list of authors on Zenodo.
 
So I hope that it would be acceptable to have Zenodo list of authors to be a superset of the authors for the publication and not necessarily matching the order.

If not acceptable: I guess we can mint a single release just for the purpose of JOSS publication with the desired listing/order.

Thank you in advance for the clarification(s). (attn @arokem @arfon )

**Comment by arfon** [comment]
> So I hope that it would be acceptable to have Zenodo list of authors to be a superset of the authors for the publication and not necessarily matching the order.

Yes this is OK. We absolutely _must_ have an archive of the software to associate with the review, and we _prefer_ that the archive metadata (authors, title etc.) match the JOSS paper. This later requirement though is a soft one so it's OK to proceed.

**Comment by yarikoptic** [comment]
We have been meticulously archiving datalad on zenodo (and pypi and debian and neurodebian and their snapshots repos) through the years and https://zenodo.org/record/5034875#.YNsmFXWYXjE is the one for recent `datalad/datalad: 0.14.6`.  Would it be sufficient (given the relaxed *prefer* for the metadata)?

**Comment by arokem** [comment]
Yes. I think this is fine.

**Comment by arokem** [comment]
@whedon set 10.5281/zenodo.5034875 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5034875" target="_blank">10.5281/zenodo.5034875</a> is the archive.

**Comment by arokem** [comment]
@openjournals/joss-eics : I believe this article is ready for your review.

**Comment by arfon** [comment]
@whedon recommend-accept

☝️ @arokem – we have this dedicated command now for flagging submissions as ready to hand over to the EiC team.

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3262 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by arfon** [comment]
@whedon recommend-accept from branch paper-joss

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1515/nf-2020-0037 is OK
- 10.5281/zenodo.4773629 is OK
- 10.5281/ZENODO.2431914 is OK
- 10.5281/ZENODO.2558512 is OK
- 10.5281/ZENODO.3874225 is OK
- 10.5281/ZENODO.3900277 is OK
- 10.5281/ZENODO.4495560 is OK
- 10.1038/sdata.2016.18 is OK
- 10.1038/s41467-021-21970-2 is OK
- 10.21203/rs.3.rs-264855/v1 is OK
- 10.1038/sdata.2016.44 is OK
- 10.31234/osf.io/w8trm is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2425

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2425, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch paper-joss 
```

**Comment by arfon** [comment]
@whedon accept deposit=true from branch paper-joss

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1410460516196229125) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2426
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03262](https://doi.org/10.21105/joss.03262)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@szorowi1, @jkanche – many thanks for your reviews here and to @arokem for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@yarikoptic – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03262/status.svg)](https://doi.org/10.21105/joss.03262)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03262">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03262/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03262/status.svg
   :target: https://doi.org/10.21105/joss.03262
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03262/status.svg)](https://doi.org/10.21105/joss.03262)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by yarikoptic** [comment]
Thank you @arfon @szorowi1, @jkanche and the last but not least @arokem for making this mundane Thursday a "once in a life time" special occasion ;)  
The idea for the "DataLad paper" was painful (we exercised a number of concepts to pursue), and the goals, format and structure of JOSS publication and review process helped us tremendously to finally make it happen!

FWIW, if might come handy for some, https://github.com/datalad/datalad-git-bug-dumps/tree/master/tools provides some extra helper we used (on top of git-bug dump) to decide on co-authors etc to invite!  No JOSS paper is expected to come for that ;)  But may be it could give ideas to establish some JOSS-tools collection of helpers on how to objectively figure out the list of co-authors to invite etc.
